front/components/data_source/RequestDataSources.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  Button,
+  DropdownMenu,
+  Modal,
+  PlusIcon,
+  TextArea,
+} from "@dust-tt/sparkle";
+import type { DataSourceType, WorkspaceType } from "@dust-tt/types";
+import { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import {
+  getConnectorProviderLogoWithFallback,
+  getDataSourceName,
+} from "@app/lib/connector_providers";
+import { isManaged } from "@app/lib/data_sources";
+import { sendRequestDataSourceEmail } from "@app/lib/email";
+import logger from "@app/logger/logger";
+
+interface RequestDataSources {
+  dataSources: DataSourceType[];
+  owner: WorkspaceType;
+}
+
+export function RequestDataSources({ dataSources, owner }: RequestDataSources) {
```

Comment:
`nit`
Sometimes `RequestDataSources` is used plural or singular can we consolidate behind one?

Reply:
removed duplicate component